pkgs/macro/lib/macro.dart

```diff
@@ -15,15 +22,278 @@ abstract interface class Macro {
   /// than relying on the user-written macro code to return it.
   ///
   MacroDescription get description;
+}
 
-  /// Generate augmentatations and diagnostics for [request].
-  ///
-  /// The code can be introspected by sending queries to [host].
-  Future<AugmentResponse> augment(Host host, AugmentRequest request);
+/// The interface for [Macro]s that can be applied to a library directive, and
+/// want to contribute new type declarations to the library.
+abstract interface class LibraryTypesMacro implements Macro {
+  FutureOr<AugmentResponse> buildTypesForLibrary(
```

Comment:
`augment` could be appropriate - because even new things like types are basically like augmentations of the library, even if they aren't augmenting a previous declaration. In a technical sense we don't call them augmentations but I don't think it's a crazy stretch :)

Reply:
I do think we should bikeshed on this later though :). Fwiw I think build is fine, but don't feel strongly either way.